### ISSUE TITLE
Template for deploying Lambda fails

### ISSUE THREAD

**Issue by crichey** [aws/aws-toolkit-jetbrains#3237]
When deploying a lambda, the template fails to deploy via sam with the error: Unrecognized resource types: [AWS::Serverless::Function]. 

Template that is created is (redacted):
`Resources:
  Function:
    Type: AWS::Serverless::Function
    Properties:
      Timeout: 300
      MemorySize: 128
      Handler: helloworld.App
      CodeUri: s3://.....redacted
      Runtime: java11
      Architectures:
      - x86_64
    Metadata:
      SamResourceId: Function
`

Adding this header manually seems to solve the problem:
`AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'`

**Comment by rli** [comment]
I thought this was pretty well documented by CloudFormation, but it would be nice to have hints in the IDE when you try to use a `AWS::Serverless` resource without the appropriate transform declared

**Comment by crichey** [comment]
Definitely some sort of hint or option to add that. The default path is to just select "create new aws lambda" from the Lambda menu and that leads to a template that isn't properly formed for deployment.

**Comment by crichey** [comment]
Yes along with some clear way to change how the template is created. If the plug-in creates a template automatically it should use the right settings.
